text/0000-edition-2021-and-beyond.md

```diff
@@ -0,0 +1,194 @@
+
+## Expected nature of editions to come
+
+We believe the Rust 2018 was somewhat exceptional in that it introduced changes to the module system that affected virtually every crate, even if those changes were almost completely automated. We expect that the changes introduced by most editions will be much more modest and discrete, more analogous to `async fn` (which simply introduced the `async` keyword), or the changes proposed by [RFC 2229] (which tweaks the way that closure captures work to make them more precise).
+
+The "size" of changes to expect is important, because they help inform the best way to ship editions. Since we expect most changes to be relatively small, we would like to design a system that allows us to judge those changes individually, without having to justify an edition by having a large number of changes combined together. Moreover, we'd like to have editions happening on a predictable cadence, so that we can take that cadence into account when designing and implementing features (i.e., so that we can try to tackle changes that may require migrations earlier, to give plenty of time).
+
+## Key ideas of edition do not change
+
+Just as with Rust 2018, we are firmly committed to the core concepts of an edition:
+
+* Crates using older editions continues to compile in the newer
+  compiler, potentially with warnings.
+* Crates using different editions can interoperate, and people can
+  upgrade to newer editions on their own schedule.
+* Code that compiles without a warning on Edition N should also
+  compile on Edition N + 1.
+* Migration between editions should generally be automated.
+* Editions make "skin-deep" changes, with all editions ultimately
+  compiling to a single common representation.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We use this section to try and convey the story that average users will need to understand.
+
+## What is a Rust edition?
+
+Every three years, we introduce a new Rust Edition. These editions are named after the year in which they occur, like Rust 2015 or Rust 2018. Each crate specifies the Rust edition that it requires in its `Cargo.toml` file via a setting like `edition = "2018"`. The purpose of editions is to give us a chance to introduce "opt-in" changes like new keywords that would otherwise have the potential to break existing code.
+
+When we introduce a new edition, we don't remove support for the older ones, so all crates continue to compile just as they ever did. Moreover, editions are fully interoperable, so there is no possibility of an "ecosystem split". This means that you can upgrade your crates to the new edition on whatever schedule works best for you.
+
+The release of a new edition is always a celebratory affair. It gives us a chance to look back at all the work that has gotten done over the last three years. The "opt-in" changes also allow us to introduce new features or syntax that would otherwise be impossible.
```

Comment:
I'm worried about this. IMHO conflation of "2018 marketing push" with "2018 compiler mode" has hurt Rust.

Celebration of last 3 years of changes at the same time as the edition switch makes it unclear what the edition switch actually does. It makes it appear like one Rust release has suddenly changed many aspects of the language in incompatible ways. Talk of migration to the new edition at the same time as a retrospective can be easily misunderstood as asking users to adopt all of the "new" features being discussed.

I suggest staggering "celebratory" and "churn" announcements in time. 

An introduction of an edition mode could be focused only on exactly what the mode does, and migration that is required. The churn should be as small as possible, so the announcement should look small too, to avoid scaring users that Rust is an unstable language that makes big incompatible changes.

A big celebratory milestone could be done at a different time. Maybe annually (Rust birthday)? Maybe a year after each edition mode? (it'd allow also reporting on edition's adoption).

Compiler switch is just an excuse for a retrospective. Rust could pick a different excuse for that.